pkg/reconciler/taskrun/validate_resources.go

```diff
@@ -140,6 +140,13 @@ func wrongTypeParamsNames(params []v1beta1.Param, matrix []v1beta1.Param, needed
 			// passed to the task that aren't being used.
 			continue
 		}
+		// This is needed to support array replacements in params. Users want to use $(tasks.taskName.results.resultname[*])
```

Comment:
This is the same in https://github.com/tektoncd/pipeline/pull/4908, once it is merged we can rebase it